src/Columns/ColumnObject.cpp

```diff
@@ -284,29 +307,26 @@ void ColumnObject::Subcolumn::insert(Field field, FieldInfo info)
     if (!is_nullable && info.have_nulls)
         field = applyVisitor(FieldVisitorReplaceNull(base_type->getDefault(), value_dim), std::move(field));
 
-    auto value_type = createArrayOfType(base_type, value_dim);
```

Comment:
Here was one of the performance issues, because many excessive `DataTypePtr` object were created.